functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/testresults/TestResult.java

```diff
@@ -44,7 +44,12 @@ public static TestResult from(String ... testResultParts) {
             throw new IllegalArgumentException(
                 "Too many arguments passed to create TestResult. Expected to get status code and name, got [" + Arrays.toString(testResultParts) + "]");
         }
-        return new TestResult(testResultParts[1], "*", Status.from(testResultParts[0]));
+        String[] classWithMethod = testResultParts[1].split("#");
+        String methodName = "*";
+        if (classWithMethod.length == 2) {
+            methodName = classWithMethod[1];
+        }
+        return new TestResult(classWithMethod[0], methodName, Status.from(testResultParts[0]));
```

Comment:
Could we create a constant to know exactly what is 0 and 1?

Reply:
constants? Do you mean something like:
```java
    private static final int POSITION_OF_TEST_CLASS_NAME = 0;
    private static final int POSITION_OF_TEST_METHOD_NAME = 1;

    public static TestResult from(String ... testResultParts) {
        if (testResultParts.length > 2) {
            throw new IllegalArgumentException(
                "Too many arguments passed to create TestResult. Expected to get status code and name, got [" + Arrays.toString(testResultParts) + "]");
        }
        String[] classWithMethod = testResultParts[1].split("#");
        String methodName = "*";
        if (classWithMethod.length == 2) {
            methodName = classWithMethod[POSITION_OF_TEST_METHOD_NAME];
        }
        return new TestResult(classWithMethod[POSITION_OF_TEST_CLASS_NAME], methodName, Status.from(testResultParts[0]));
    }
```
This doesn't seem to be very meaningfull from my point of view - this also brings a complication as there is another array `testResultParts`...

To say what should be inside of the array, I named it: `classWithMethod`
in addition, I assign the `classWithMethod[1]` to `methodName` which says what is expected there. 
I could also assign the first variable: 
```java
        String className = classWithMethod[0];
        if (classWithMethod.length == 2) {
            methodName = classWithMethod[1];
        }
        return new TestResult(className, methodName, Status.from(testResultParts[0]));
```
which is slightly redudant, but I can do it

Reply:
I think the first approach is good  setting the const as static ints with a name, but maybe creating a method for the if might be good and help to understand the method so instead of `clasWithMethod.length==2` do something like `if(classContainsMethod(classWithMethod))`

Reply:
using the constants for the positions was irony... I would need to name it something like:
`POSITION_OF_TEST_CLASS_NAME_IN_CLASS_WITH_METHOD_ARRAY` because there is another array `testResultParts` which is something that I don't really like...

Does this make more sense to you?
```java
    public static TestResult from(String ... testResultParts) {
        if (testResultParts.length > 2) {
            throw new IllegalArgumentException(
                "Too many arguments passed to create TestResult. Expected to get status code and name, got [" + Arrays.toString(testResultParts) + "]");
        }

        String className;
        String methodName;

        if (expectedTestContainsSpecificMethodDefinition(testResultParts[1])) {
            String[] classWithMethod = testResultParts[1].split("#");
            className = classWithMethod[0];
            methodName = classWithMethod[1];

        } else {
            className = testResultParts[1];
            methodName = "*";
        }
        return new TestResult(className, methodName, Status.from(testResultParts[0]));
    }

    private static boolean expectedTestContainsSpecificMethodDefinition(String expectedTest) {
        return expectedTest.contains("#");
    }
```

Reply:
ah yeah now better I guess

Reply:
Done